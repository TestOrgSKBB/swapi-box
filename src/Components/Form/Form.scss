$dark-grey: rgba(32, 37, 40, .8);
$card-background: rgba(145, 145, 145, 0.8);
$element-background: rgba(51, 51, 51, .5);

form {
  display: flex;
  flex-direction: column;
  border-radius: 3em;
}

.form__element {
  border-radius: 4px;
  font-size: 1.5em;
  font-weight: bold;
  height: 8vh;
  margin: 5% 5% 0% 5%;
  opacity: .7;
  padding-left: 7px;
}

#form__div--container {
  width: 60%;
  margin-left: 20%;
  margin-top: 6%;
}

.form__button {
  border-radius: 4px;
  cursor: pointer;
  font-size: 2em;
  font-weight: bold;
  height: 8vh;
  width: 90%;
  color: $card-background;
  &:hover {
    background-color: $element-background;
    color: white;
  }
}

.form__button:disabled {
  cursor: not-allowed;
  background-color: white;
  color: $card-background
}

.form__button--link {
  margin: 5% 5% 5% 5%;
  height: 8vh;
  width: 100%;
}

.form-title {
  color: #919192;
  font-size: 3em;
  position: absolute;
  top: 0;
  margin-left: 18%;
}

