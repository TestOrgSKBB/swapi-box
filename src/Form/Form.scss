form {
  display: flex;
  flex-direction: column;
}

.form__element {
  margin: 5% 5% 0% 5%;
  height: 8vh;
  opacity: .7;
}

#form__div--container {
  border: 2px solid #BFCAC3;
  width: 60%;
  margin-left: 20%;
  margin-top: 6%;
}

.form__button {
  // margin-bottom: 5%;
  width: 90%;
  height: 8vh;
  cursor: pointer;
}

.form__button:disabled {
  cursor: not-allowed;
}

.form__button--link {
  margin: 5% 5% 5% 5%;
  height: 8vh;
  width: 100%;
}

