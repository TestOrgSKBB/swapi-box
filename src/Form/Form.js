import React, { Component } from 'react';
import { NavLink } from 'react-router-dom'
import $ from 'jquery';
import './Form.scss';
import PropTypes from 'prop-types';

class Form extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      quote: '',
      rank: '',
      isComplete: false,
    };
  };

  handleChange = (e) => {
    this.setState({[e.target.name]: e.target.value})
    const isNameGood = $('#form__first-name').val() !== '';
    const isQuoteGood = $('#form__fav-quote').val() !== '';
    const isRankGood = $('select').val() !== 'Rank';

    if(isNameGood && isQuoteGood && isRankGood) {
      this.setState({ isComplete: true })
    } else {
      this.setState({ isComplete: false })
    };
  };

  submitUserInfo = () => {
    const name = `Name: ${$('#form__first-name').val()}`;
    const quote = `Quote: ${$('#form__fav-quote').val()}`;
    const rank = `Rank: ${$('select').val()}`;
    this.props.updateState({name, quote, rank});
  };

  render() {
    return(
      <div id="form__div--container">
      <h1 className="form-title">Star Wars: Trivia</h1>
      <form>
        <input
          id="form__first-name" 
          className="form__element form__input"
          type="text" 
          placeholder="Enter Name"
          value={ this.state.name }
          name="name"
          onChange={this.handleChange}
          />
        <input 
          id="form__fav-quote"
          className="form__element from__input"
          type="text" 
          placeholder="Enter Favorite Star Wars Quote"
          value={ this.state.quote }
          name="quote"
          onChange={this.handleChange}
          />
        <select className="form__element" name="rank" value={this.state.rank} onChange={this.handleChange}>
          <option defaultValue="rank">Rank</option>
          <option name="rank" value="beginner">Beginner</option>
          <option name="rank" value="intermediate">Intermediate</option>
          <option name="rank" value="expert">Expert</option>
        </select>
        <NavLink className="form__button--link" to='/movies'>
          {!this.state.isComplete && <button 
            className="form__button" 
            disabled  
          >Submit</button>}
          {this.state.isComplete && <button
            className="form__button"
            onClick={this.submitUserInfo}
          >Submit</button>}
        </NavLink>
      </form>
      </div>
    )
  };
};

Form.propTypes = {
  movies: PropTypes.array,
  handleChange: PropTypes.func,
  submitUserInfo: PropTypes.func
};

export default Form;