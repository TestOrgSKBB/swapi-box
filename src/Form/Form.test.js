import React from 'react';
import { shallow } from 'enzyme'
import Form from './Form';
import { isTSAnyKeyword, exportAllDeclaration } from '@babel/types';

describe('Form', () => {
  let wrapper
  
  beforeEach(() => {
    wrapper = shallow(<Form />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

})