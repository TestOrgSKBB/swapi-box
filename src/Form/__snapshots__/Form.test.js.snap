// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot 1`] = `
<div
  id="form__div--container"
>
  <form>
    <input
      className="form__element form__input"
      name="name"
      onChange={[Function]}
      placeholder="Enter Name"
      type="text"
      value=""
    />
    <input
      className="form__element from__input"
      name="quote"
      onChange={[Function]}
      placeholder="Enter Favorite Star Wars Quote"
      type="text"
      value=""
    />
    <select
      className="form__element"
      name="rank"
      onChange={[Function]}
      value=""
    >
      <option
        defaultValue="rank"
      >
        Rank
      </option>
      <option
        name="rank"
        value="beginner"
      >
        Beginner
      </option>
      <option
        name="rank"
        value="intermediate"
      >
        Intermediate
      </option>
      <option
        name="rank"
        value="expert"
      >
        Expert
      </option>
    </select>
    <button
      className="form__element form__button"
      disabled={true}
      onClick={[Function]}
    >
      Submit
    </button>
  </form>
</div>
`;
